lib/saml.js

```diff
@@ -182,31 +182,54 @@ SAML.prototype.certToPEM = function (cert) {
 SAML.prototype.checkSAMLStatus = function (xmlDomDoc) {
 	var status = {StatusCodeValue:null, StatusMessage:null, StatusDetail:null}
 
-	var statusCodeValueNode = xmlCrypto.xpath(xmlDomDoc, "//*[local-name(.)='StatusCode']")[0];
-	if(statusCodeValueNode)
-	{
+	try{
+    var statusCodeValueNode = xmlCrypto.xpath(xmlDomDoc, "//*[local-name(.)='StatusCode']")[0];
```

Comment:
`statusCodeValueNode` should be defined at the top of the function for proper scoping.